pkg/skaffold/build/local/custom.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package local
+
+import (
+	"context"
+	"io"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/custom"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/pkg/errors"
+)
+
+func (b *Builder) buildCustom(ctx context.Context, out io.Writer, artifact *latest.Artifact, tag string) (string, error) {
+	customArtifactBuilder := custom.NewArtifactBuilder(b.pushImages, b.localDocker.ExtraEnv())
+
+	if err := customArtifactBuilder.Build(ctx, out, artifact, tag); err != nil {
+		return "", errors.Wrap(err, "building custom artifact")
+	}
+
+	if b.pushImages {
+		return docker.RemoteDigest(tag, b.insecureRegistries)
```

Comment:
So the custom build script is responsible for pushing. We pass in `pushImages` to the build script so that it knows if the image is expected to exist remotely or locally.

Reply:
no my question was around "skaffold actually pushing an image here instead of retrieving it". Since custom artifact builder is local builder the build script shd have build it in local docker.
But i think after looking at what other builder do there are builder which directly build into registry.
So i change my mind.